src/index.test.ts

```diff
@@ -63,6 +63,151 @@ describe('basic go usage', () => {
+    const res = await timeoutGo(operations.successFn, { timeoutMs: 5 });
+    expect(res).toEqual(fail(new Error('Operation timed out after final retry')));
+  });
+});
+
+describe('basic retryTimeoutGo usage', () => {
+  const operations = {
+    successFn: () =>
+      new Promise((res) =>
+        setTimeout(() => {
+          res(2);
+        }, 20)
+      ),
+    errorFn: () => new Promise((_res, rej) => setTimeout(() => rej(new Error('Computer says no')), 20)),
+  };
+
+  it('resolves successful asynchronous functions', async () => {
+    const res = await retryTimeoutGo(operations.successFn, { timeoutMs: 50, retries: 3 });
+    expect(res).toEqual(success(2));
+  });
+
+  it('resolves unsuccessful asynchronous functions', async () => {
+    const res = await retryTimeoutGo(operations.errorFn, { timeoutMs: 50, retries: 3 });
+    expect(res).toEqual(fail(new Error('Computer says no')));
+  });
+
+  it('retries and resolves successful asynchronous functions', async () => {
+    jest
+      .spyOn(operations, 'successFn')
+      .mockRejectedValueOnce(new Error('Operation timed out, retries left: 1'))
+      .mockRejectedValueOnce(new Error('Operation timed out after final retry'));
+
+    const res = await retryTimeoutGo(operations.successFn, { timeoutMs: 100, retries: 3 });
+    expect(operations.successFn).toHaveBeenCalledTimes(3);
+    expect(res).toEqual(success(2));
+  });
+
+  it('retries and resolves unsuccessful asynchronous functions', async () => {
+    jest
+      .spyOn(operations, 'errorFn')
+      .mockRejectedValueOnce(new Error('Operation timed out, retries left: 1'))
+      .mockRejectedValueOnce(new Error('Operation timed out after final retry'));
+
+    const res = await retryTimeoutGo(operations.errorFn, { timeoutMs: 100, retries: 2 });
+    expect(operations.errorFn).toHaveBeenCalledTimes(3);
+    expect(res).toEqual(fail(new Error('Computer says no')));
+  });
+
+  it('retries and resolves unsuccessful timed out asynchronous functions', async () => {
+    jest
+      .spyOn(operations, 'successFn')
+      .mockRejectedValueOnce(new Error('Operation timed out, retries left: 2'))
+      .mockRejectedValueOnce(new Error('Operation timed out, retries left: 1'))
+      .mockRejectedValueOnce(new Error('Operation timed out after final retry'));
+
+    const res = await retryTimeoutGo(operations.successFn, { timeoutMs: 100, retries: 2 });
+    expect(operations.successFn).toHaveBeenCalledTimes(3);
+    expect(res).toEqual(fail(new Error('Operation timed out after final retry')));
+  });
+
+  it('retries and resolves unsuccessful timed out functions', async () => {
+    const res = await retryTimeoutGo(
+      () =>
+        new Promise((res) =>
+          setTimeout(() => {
+            res(2);
+          }, 200)
+        ),
+      { timeoutMs: 50, retries: 3 }
+    );
+    expect(res).toEqual(fail(new Error('Operation timed out after final retry')));
+  });
+});
```

Comment:
This is where you should check whether the function was called the correct number of times and then it timeouted.